cmake/profiles/develop.cmake

```diff
@@ -46,6 +49,7 @@ if(MBED_TOOLCHAIN STREQUAL "GCC_ARM")
 elseif(MBED_TOOLCHAIN STREQUAL "ARM")
     list(APPEND c_compile_options
         "-std=gnu11"
+        "-Os"
```

Comment:
Shouldn't this be `-Oz` as this is the Arm Compiler?

Reply:
It was decided to be Os some time ago. I am not changing flags in this PR, just matching what we have on master.

Reply:
OK, it feels like this is incorrect and should be fixed separately. Is there any documentation on the decision? I'd love to read and ADR to get the context behind the decision, or at the very least a commit message.

Commit 98b4768434e0 "Initial toolchain class for ARMC6", which sets the flags in this way, does not explain why Os is used in develop and Oz is used in release.

Reply:
It might be worth creating a new issue to review the optimization flags (why they are not consistent), it will take a bit of time to find these in the history or comments..

Reply:
If only we had some `doc/adr` folder... ;)